prog.go

```diff
@@ -155,14 +155,32 @@ func (p *BPFProg) SetAttachTarget(attachProgFD int, attachFuncName string) error
 	return nil
 }
 
-// TODO: fix API to return error
+// Deprecated: use BPFProg.SetExpectedProgramType() instead.
 func (p *BPFProg) SetProgramType(progType BPFProgType) {
-	C.bpf_program__set_type(p.prog, C.enum_bpf_prog_type(int(progType)))
+	C.bpf_program__set_type(p.prog, C.enum_bpf_prog_type(C.int(progType)))
```

Comment:
Now you can just call the new API here:

```suggestion
	_ = p.SetType(progType)
```